File: greet1.ex
defmodule Greet1 do
  def greet(:male, name) do
    IO.write("Hello, Mr. #{name}!")
  end
  def greet(:female, name) do
    IO.write("Hello, Mrs. #{name}!")
  end
  def greet(_, name) do
    IO.write("Hello, #{name}!")
  end

  def old_enough(x) when x >= 16 do
    true
  end
  def old_enough(_) do
    false
  end
  def right_age(x) when x >= 16 and x <= 104 do
    true
  end
  def right_age(_) do
    false
  end
  def wrong_age(x) when x < 16 or x > 104 do
    true
  end
  def wrong_age(_) do
    false
  end
end

File: greet1.erl
-module(greet1).
-compile(export_all).
% -export([greet/2]).

greet(male, Name) ->
    io:format("Hello, Mr. ~s!", [Name]);
greet(female, Name) ->
    io:format("Hello, Mrs. ~s!", [Name]);
greet(_, Name) ->
    io:format("Hello, ~s!", [Name]).

old_enough(X) when X >= 16 -> true;
old_enough(_) -> false.

right_age(X) when X >= 16, X =< 104 -> true;
right_age(_) -> false.

wrong_age(X) when X < 16; X > 104 -> true;
wrong_age(_) -> false.